app/routes.py

```diff
@@ -1 +1,205 @@
+def update_task(task_id):
+    chosen_task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    try:
+        chosen_task.title = request_body["title"]
+        chosen_task.description = request_body["description"]
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    db.session.commit()
+    
+    return jsonify({"task": chosen_task.to_json()})
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    chosen_task = validate_model(Task, task_id)
+    
+    db.session.delete(chosen_task)
+    db.session.commit()
+
+    return jsonify({"details": f"Task {task_id} \"{chosen_task.title}\" successfully deleted"})
+
+def call_slack_api(text):
+    url = "https://slack.com/api/chat.postMessage"
+    headers = {"Authorization": os.environ.get("SLACK_TOKEN")}
+    query_params = {
+        "channel": "task-notifications",
+        "text": text
+        }
+
+    response = requests.post(url, headers=headers, params=query_params)
+
+    return response
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    chosen_task = validate_model(Task, task_id)
+    chosen_task.completed_at = date.today()
+    call_slack_api(f"Someone just completed the task {chosen_task.title}")
+
+    db.session.commit()
+
+    chosen_task_dict = chosen_task.to_json()
+    chosen_task_dict["is_complete"] = True
+    return jsonify({"task": chosen_task_dict})
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    chosen_task = validate_model(Task, task_id)
+    chosen_task.completed_at = None
+
+    db.session.commit()
+
+    return jsonify({"task": chosen_task.to_json()})
+
+
+# **************************** CRUD ROUTES FOR GOALS *****************************************
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    if request_body:
+        new_goal = Goal.from_json(request_body)
+    else:
+        return jsonify({"details": "Invalid data"}), 400
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_json()}), 201
+
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals = Goal.query.all()
+
+    goals_response = [goal.to_json() for goal in goals]
+
+    return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    chosen_goal = validate_model(Goal, goal_id)
+
+    return jsonify({"goal": chosen_goal.to_json()})
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    chosen_goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    try:
+        chosen_goal.title = request_body["title"]
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400
```

Comment:
Nice use of try-except, but again a more meaningful error message is appropriate.